x/auth/tx/legacy_amino_json.go

```diff
@@ -0,0 +1,61 @@
+package tx
+
+import (
+	"fmt"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/auth/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	signingtypes "github.com/cosmos/cosmos-sdk/types/tx/signing"
+	"github.com/cosmos/cosmos-sdk/x/auth/signing"
+)
+
+// signModeLegacyAminoJSONHandler defines the SIGN_MODE_LEGACY_AMINO_JSON SignModeHandler
+type signModeLegacyAminoJSONHandler struct{}
+
+func (s signModeLegacyAminoJSONHandler) DefaultMode() signingtypes.SignMode {
+	return signingtypes.SignMode_SIGN_MODE_LEGACY_AMINO_JSON
+}
+
+func (s signModeLegacyAminoJSONHandler) Modes() []signingtypes.SignMode {
+	return []signingtypes.SignMode{signingtypes.SignMode_SIGN_MODE_LEGACY_AMINO_JSON}
+}
+
+const aminoNonCriticalFieldsError = "protobuf transaction contains unknown non-critical fields. This is a transaction malleability issue and SIGN_MODE_LEGACY_AMINO_JSON cannot be used."
+
+func (s signModeLegacyAminoJSONHandler) GetSignBytes(mode signingtypes.SignMode, data signing.SignerData, tx sdk.Tx) ([]byte, error) {
+	if mode != signingtypes.SignMode_SIGN_MODE_LEGACY_AMINO_JSON {
+		return nil, fmt.Errorf("expected %s, got %s", signingtypes.SignMode_SIGN_MODE_LEGACY_AMINO_JSON, mode)
+	}
+
+	protoTx, ok := tx.(*builder)
+	if !ok {
+		return nil, fmt.Errorf("can only handle a protobuf Tx, got %T", tx)
+	}
+
+	if protoTx.txBodyHasUnknownNonCriticals {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, aminoNonCriticalFieldsError)
+	}
+
+	body := protoTx.tx.Body
+
+	if len(body.ExtensionOptions) != 0 || len(body.NonCriticalExtensionOptions) != 0 {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest,
+			"SIGN_MODE_LEGACY_AMINO_JSON does not support protobuf extension options.")
+	}
+
+	if body.TimeoutHeight != 0 {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest,
+			"SIGN_MODE_LEGACY_AMINO_JSON does not support timeout height.")
+	}
```

Comment:
@alexanderbez note that `TimeoutHeight` is being now rejected by `SIGN_MODE_LEGACY_AMINO_JSON` (because otherwise it's a malleability issue) so if you are adding it to `StdTx` in #6089, you'll need to remove these lines.

Reply:
Thank's for pointing that out!